clash-prelude/src/Clash/Num/Zeroing.hs

```diff
@@ -1,11 +1,12 @@
-{-
-Copyright   : (C) 2021, QBayLogic B.V.
+{-|
+Copyright   : (C) 2021-2024, QBayLogic B.V.
 License     : BSD2 (see the file LICENSE)
 Maintainer  : QBayLogic B.V. <devops@qbaylogic.com>
 -}
 
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
```

Comment:
Why does importing a module cause an additional language pragma?

Reply:
It's because `Finite` also has been added to the `deriving newtype` list and the default implementation of `Finite` requires `TypeFamilies`. The `deriving newtype` strategy does not use the default implementation tough, which is why the dependency technically is not necessary here. This seems to be fixed in GHC 9.4 and later, but GHC 9.2 and older still required the language extension to be enabled.

We could add a preprocessor define to only enable the extension for the old GHCs, but then we also need to enable `CCP` for any compiler version, so it's a matter of taste in the end. A CPP define only makes sense if we plan to clean up the code as soon as we drop support for GHC 9.2 and older, because it makes it easier identify the legacy parts.